cmd/kyma/alpha/create/module/module.go

```diff
@@ -12,12 +12,14 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/kyma-project/cli/internal/cli"
+	kustomizeSetup "github.com/kyma-project/cli/internal/cli/setup/kustomize"
 	"github.com/kyma-project/cli/pkg/module"
 )
 
 type command struct {
-	opts *Options
 	cli.Command
+	opts *Options
+	kustomizeSetup.ExtendCmd
```

Comment:
why not directly use the function `SetupKustomize` in `kustomize` package, instead you define an `ExtendCmd`?

Reply:
OK, thanks for explain, the purpose to composite command is understandable. But I actually thinking from different point of view:
1. If we are talking about the mixin, the original idea for mixin is want to let a class (e.g: class A) which contains methods defined by other classes (e.g: class B) without inheritance (so called composition). But why we need to declare class B inside class A first? Because in other common OO language like JAVA, there is no way you can simply define a method without belonging to a class. But golang already provide us this ability, just simply import package, then you can immediately use public method inside those package, which means the composite is naturally achievable in package level. So we don't need to declare an empty `ExtendCmd` struct in order to use method. To me, declare a class (structs) to achieve mixin is just because some language lack of the ability to directly use method, so it's a compromised indirect way.

2. Regarding readability, maybe by read the structs, I mean `command` struct, we can briefly know this command contains kustomize setup command, or later other commands. But if you want to find in the code where is it actually used, simple have no idea, because those `ExtendCmd` is implied by the `command`, every place you use the actually method declared in `ExtendCmd`, you basically use it with `command`, like current case, instead of calling `ExtendCmd.SetupKustomize`, we directly call `cmd.SetupKustomize`, to me, it's against improving readability. Instead, by import package, I can use different package name to call this method.

Reply:
@Tomasz-Smelcerz-SAP I think @ruanxin got a good point here. IMO this seems a bit like over engineering. We already have the public available kustomize Setup function in [here](https://github.com/kyma-project/cli/blob/f94071d921d50b73316483c317ce7edbeb20858d/internal/kustomize/kustomize.go#L70). This could be directly called instead of having to extend it with a new struct.

I totally get your point, if having a short declaration including all the necessary steps to have kind of like a `shorter` summary of what the command is doing, just by reading the struct. But for this I think we need to come up with a proper interface solution, which could take more effort, since we then also need to adapt all the other commands.

I would suggest to go now for the simplest solution, meaning calling the kustomize setup step directly from the package.